### PR TITLE
fixing MacroCompiler, context and arguments aren't merged

### PR DESCRIPTION
**Pull request by oleg-andreyev**
#96

Ref.: https://github.com/twigphp/Twig/blob/v1.16.2/lib/Twig/Node/Macro.php#L55 (more readable logic)

### CODE REVIEW

**Comment by hnrysmth** [comment]
Oooh this sure sounds like it could be a bug fix. Is it? What sort of circumstances does this bug happen in? What effects does it have?

If you could briefly explain some stuff about it, and I guess also include some kind of unit test or integration (or both, if you like) for the fix, this would be trivial to merge 👍

**Comment by oleg-andreyev** [comment]
Don't quite remember the case, but iirc it was related to passing arguments to macro, before this PR they were ignored.

File: tests/TwigJs/Tests/Fixture/generated/macro.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+ * @param {*} opt_arg2
+ * @return {string}
+ */
+macro.prototype.getmacro_test = function(opt_arg1, opt_arg2) {
+    var context = twig.extend({}, {"arg1":opt_arg1,"arg2":opt_arg2}, this.env_.getGlobals());
+
+    var sb = new twig.StringBuffer;
+    // line 2
+    sb.append("    ");
+    context["arg3"] = ((("arg2" in context)) ? (twig.filter.def(opt_arg2, "")) : (""));
```
Comment (oleg-andreyev) [inline tests/TwigJs/Tests/Fixture/generated/macro.js]: `{% set arg3 = arg2|default('') %}`

File: tests/TwigJs/Tests/Fixture/generated/macro.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    // line 3
+    sb.append("    ");
+    sb.append(twig.filter.escape(this.env_, opt_arg1, "html", null, true));
+    sb.append("\n    ");
+    // line 4
+    if ((!twig.empty(("arg3" in context ? context["arg3"] : null)))) {
```
Comment (oleg-andreyev) [inline tests/TwigJs/Tests/Fixture/generated/macro.js]: `{% if arg3 is not empty %}{{ arg3 }}{% endif %}`

**Comment by oleg-andreyev** [comment]
@hnrysmth can you take a look?

**Pull request closed without merging**